lib/models/database/user_comment/user_comment_firestore.dart

```diff
@@ -0,0 +1,44 @@
+import "package:cloud_firestore/cloud_firestore.dart";
+import "package:proxima/models/database/user_comment/user_comment_data.dart";
+import "package:proxima/models/database/user_comment/user_comment_id_firestore.dart";
+
+/// This class represents a comment from the user in firestore.
+class UserCommentFirestore {
+  static const String userCommentSubCollectionName = "userComments";
+
+  // The id of the firestore document that keeps the reference to the comment.
+  // Not the id of the comment itself.
+  final UserCommentIdFirestore id;
```

Comment:
By humble curiosity, why did you choose to use a different ID as the `UserCommentIdFirestore`? Wouldn't it make sense to directly user `CommentIdFirestore` as id?
I feel like using a `CommentIdFirestore` as id would decrease the complexity by a non-negligible factor: you could remove it from `UserCommentData` and delete the class `UserCommentIdFirestore`.

Reply:
At first, I thought it would make more sense to have them separated because all the comments are not in the same firestore collection.
But looking back at it, it adds quite a lot of complexity for not much thus I think that you are right.
Thanks for pointing it out !

Fix in 57a4b8d3caea5a2bbfa420938186a05afae385e2.